### PR TITLE
mgr: TTL cache implementation

### PR DESCRIPTION
**Pull request by alfonsomthd**
Fixes: https://tracker.ceph.com/issues/53561

**TODO**

- [x] Remove injector
- [x] Remove cli lightweight
- [x] Retry benchmarking again when the other cli pr is merged

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
yes please, let's put it into master.

**Comment by epuertat** [comment]
jenkins retest this please

**Comment by epuertat** [comment]
jenkins test api

**Comment by epuertat** [comment]
jenkins test dashboard cephadm

File: src/common/options/global.yaml.in [hunk 1]
```diff
@@ -6086,4 +6086,31 @@ options:
...
+- name: mgr_inject_num_osds
+  type: int
+  level: dev
+  desc: Number of osds to inject in the mgr module
+  default: 0
+  flags:
+  - runtime
+  with_legacy: true
+  services:
+  - mgr
```
Comment (epuertat) [inline src/common/options/global.yaml.in:6109]: @pereman2 we talked about not merging the fake OSD injector?

File: src/test/mgr/CMakeLists.txt [hunk 2]
```diff
@@ -1,16 +1,21 @@
 # unittest_mgr_mgrcap
 add_executable(unittest_mgr_mgrcap
-  test_mgrcap.cc
-  $<TARGET_OBJECTS:mgr_cap_obj>)
+	test_mgrcap.cc
```
Comment (epuertat) [inline src/test/mgr/CMakeLists.txt:3]: [tabs, arghh](https://www.youtube.com/watch?v=SsoOG6ZeyUI)!

**Review by epuertat — CHANGES_REQUESTED** [review]

File: src/test/mgr/CMakeLists.txt (hunk 2 above)
Comment (pereman2) [reply src/test/mgr/CMakeLists.txt:3]: it looks like I have a command in my config doing this xD, I won't let you down again sensei.

File: src/common/options/global.yaml.in (hunk 1 above)
Comment (pereman2) [reply src/common/options/global.yaml.in:6109]: yes, it should be removed

**Comment by alfonsomthd** [comment]
Shaman build:
https://shaman.ceph.com/builds/ceph/wip-alfonsomthd-testing-2021-12-07-0929/c20ac5d5e7b17d3db4342b0a42115bc7adb6f8b6/

**Comment by pereman2** [comment]
I reran the benchmarks and got some interesting results. 

**Using python native json library**
nativejson

**Using orjson python library to deserialize only**
simple

**Comment by pereman2** [comment]
I ran them again with 5k iterations, no clear improvement with orjson. 
simple

**Comment by pereman2** [comment]
https://shaman.ceph.com/builds/ceph/wip-pereman2-testing-2021-12-15-1507/5f2f524271a270645a6094fec557433348b1dd7d/

**Comment by pereman2** [comment]
jenkins test make check

**Comment by pereman2** [comment]
https://pulpito.ceph.com/pdiazbou-2022-01-03_12:05:09-rados:mgr-wip-pereman2-testing-2021-12-15-1507-distro-basic-smithi/
20 passs 8 dead (because of rhel)

**Comment by pereman2** [comment]
jenkins test make check

**Comment by pereman2** [comment]
jenkins test make check

**Comment by pereman2** [comment]
jenkins test make check

**Comment by pereman2** [comment]
jenkins test make check

**Review by pereman2 — APPROVED** [review]
good job Alfonso!

**Review by epuertat — APPROVED** [review]
LGTM! After keeping this soaking for more than half a year, it's time to get it merged!

**Pull request merged**